### ISSUE TITLE
ContrastiveTrainer "probe" not useful as implemented

### ISSUE THREAD

**Issue by martin-gorner** [keras-team/keras-cv#1911]
Source file https://github.com/keras-team/keras-cv/blob/master/keras_cv/training/contrastive/contrastive_trainer.py

A "linear probe" is a linear classifier trained on top of a frozen (typically contrastively pretrained) backbone. It is usually trained after the contrastive pretraining, to evaluate the quality of the pre-training (example in [this Medium article](https://shairozsohail.medium.com/contrastive-representation-learning-a-comprehensive-guide-part-2-scoring-representations-bedd28eddb30#6c16))

A hack that is sometimes used is to train the linear probe during contrastive pre-training, to obtain an in-training "accuracy" curve which helps debug convergence problems. This in-training linear probe can be less accurate.
* This hack is implemented [here for example](https://github.com/beresandras/contrastive-classification-keras/blob/c32077d073c74cd741a456415035a77b5db1df77/models.py#L117).
* It is also implemented in KerasCV's ContrastiveTrainer [here](https://github.com/keras-team/keras-cv/blob/f05494c1057c95cbf44abac3238afcf262a50431/keras_cv/training/contrastive/contrastive_trainer.py#L258).

The important difference between the two implementations is the dataset. Contrastive training uses an unlabeled (typically large) dataset. Training the linear probe requires a labeled (typically smaller) dataset. Unfortunately, the KerasCV implementation does not allow passing a mixed dataset with both labeled and unlabeled images. You can either train, using the probe, on a labeled dataset, which defeats the purpose of contrastive training, or train on an unlabled dataset and not use the probe param.

**Comment by martin-gorner** [comment]
Note: it is debatable wether a contrastive trainer should be in KerasCV. A much better match for it [TensorFlow Similarity](https://github.com/tensorflow/similarity) which actually already implements a contrastive trainer in Keras.

**Comment by innat** [comment]
> Note: it is debatable wether a contrastive trainer should be in KerasCV. A much better match for it [TensorFlow Similarity](https://github.com/tensorflow/similarity) which actually already implements a contrastive trainer in Keras.

@martin-gorner 
[TensorFlow-Addons](https://www.tensorflow.org/addons) have stopped development and deprecated. And TF-Similarity might have many components that sooner or later will conflict with keras-cv/keras-nlp.

- what if TF-Similarity is also depreciated in favor of keras-cv/nlp?
- what if? https://github.com/tensorflow/similarity/issues/261
- purpose of `keras_cv/training/*`! https://github.com/keras-team/keras-cv/discussions/1734

**Comment by sachinprasadhs** [comment]
Thanks for reporting the issue! We have consolidated the development of KerasCV into the new **KerasHub** package, which supports image, text, and multi-modal models. Please read [the announcement](https://github.com/keras-team/keras-hub/issues/1831). `KerasHub` will support all the core functionality of `KerasCV`.

`KerasHub` can be installed with `!pip install -U keras-hub`. Documentation and guides are available at [keras.io/keras_hub](https://keras.io/keras_hub/).

With our focus shifted to KerasHub, we are not planning any further development or releases in KerasCV. If you encounter a KerasCV feature that is missing from KerasHub, or would like to propose an addition to the library, please file [an issue with KerasHub](https://github.com/keras-team/keras-hub/issues).
